test/fixtures/modules/values-in-selectors.css

```diff
@@ -0,0 +1,14 @@
+@value other from './values.css';
+@value button-class: btn;
+
+:import('imported-simple.css') {
+  other-child: imported-simple;
+}
+/* At the moment local-by-default  doesn't understand values .btn would get hashed */
```

Comment:
hm, i think we should solve this before release new version

Reply:
I think it'd be worth deferring that. 

- it's not obviously wrong behavior (maybe not a bug)
- it's not simple to solve well
- it's not a common use case for this and easily worked around

Reply:
@jquense hm, can you clarify with examples, i am afraid we can break many code with new release

Reply:
I don't think this is a bug necessarily, and changing it would break _more_ people than this release currently will.

here is the current versions behavior:

```css
@value foo: blue;

// .foo become ._22ETk3ZhhvjVe-4dAULcsF
.foo { 
 display: block
}
```

That will stay the same. If explicitly want `.foo` to be replaced by `.blue` you need to do `:global(.foo)`. current code without `:global` won't change behavior

Reply:
hm, based on https://github.com/css-modules/icss#import we should replace `foo` without `:global`, or maybe I misunderstood you. 

When modules were enabled we replaces values in selectors by default?

Reply:
oh, i see, i think it is invalid behavior, when i am using `:global` i want selector will be global without any processing

Reply:
`foo` isn't an `:import` (`other` is and is replaced) so i don't think so. To be clear what's happening is we are replacing `.foo` with `.blue`, and then `.blue` is hashed and exported.

> When modules were enabled we replaces values in selectors by default?

we replace _imports_ in selectors, that includes imported values not just imported classes. and yes it's on by default, since that's the correct behavior according to the ICSS spec


There are really two related but slightly different changes. `@value` replacement, which isn't defined by the spec. and proper `:import` replacement,  which is specced out.

the change we made to `local-by-default` is enable imports to be used in selectors without them breaking, but that plugin has no idea what a `@value` is, and probably shouldn't.

overall this is so complex because the css-module **default** is bad and involves a whole processing step to wrap classes in `:local` instead of making that the default already

Reply:
> when i am using :global i want selector will be global without any processing

That's not clear to me. `global` just means "do not hash" it doesn't mean don't do any processing.

again tho this is really hard to change unless `local-by-default` plugins knows about `@value`s, you'd have to combine the two plugins for that, and that's a bit weird b/c `@value` isn't even a specced css-module thing, its just a plugin that css-loader happens to use

Reply:
fair enough, we need update docs in this PR for other developers

Reply:
awesome yeah i forgot about the docs. will update

Reply:
:+1: